os/lib/csprng.c

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2013, Hasso-Plattner-Institut.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. Neither the name of the Institute nor the names of its contributors
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE INSTITUTE AND CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE INSTITUTE OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ *
+ * This file is part of the Contiki operating system.
+ *
+ */
+
+/**
+ * \file
+ *         An OFB-AES-128-based CSPRNG.
+ * \author
+ *         Konrad Krentz <konrad.krentz@gmail.com>
+ */
+
+/**
+ * \addtogroup csprng
+ * @{
+ */
+
+#include "lib/csprng.h"
+#include "lib/aes-128.h"
+#include "sys/cc.h"
+#include <string.h>
+
+/* Log configuration */
+#include "sys/log.h"
+#define LOG_MODULE "CSPRNG"
+#define LOG_LEVEL LOG_LEVEL_NONE
+
+static struct csprng_seed seed;
+static unsigned read_state_bytes;
+static int seeded;
+
+/*---------------------------------------------------------------------------*/
+void
+csprng_feed(struct csprng_seed *new_seed)
+{
+  uint8_t i;
```

Comment:
Just a heads up: We have moved from c89, so you are free to define the variable where you initialize. I personally prefer that style, but I don't believe we enforce either way, so feel free to make changes or leave as is.

Reply:
Putting declarations first complies with [Contiki's code style document ](https://docs.contiki-ng.org/en/develop/doc/project/Code-style.html) and much of the code in `os/` ...

Reply:
I was not aware the style document actually had a "should always", I am fairly certain we don't enforce either way, so I will see if we won't change that wording. In any case, the current style in this PR is fine.